seeds/_sector/dex/curvefi_optimism_base_trades_seed.csv

```diff
@@ -1,18 +1,21 @@
+optimism,curve,1,2023-01-30,0x76780eb40844a0fc1cbaa16a30d8098d55bec4d774cfee1a7baed9c8dd4f6a92,8,0x8c6f28f2f1a3c87f0f938b96d27520d9751ec8d9,0x94b008aa00579c1307b0ef2c499ad98a8ce58e58,71367878,79900141472128748,78688144336772701
+optimism,curve,1,2023-01-30,0x72f451639593a62c0ec916d20de947b11d15f2ae5075452ddf08d68f77b54e87,8,0x8c6f28f2f1a3c87f0f938b96d27520d9751ec8d9,0x94b008aa00579c1307b0ef2c499ad98a8ce58e58,71362505,99875991930621765,98360937855263771
+optimism,curve,1,2023-01-31,0xa70d826c5b3b5afbe9585e15f1e58c75cbcc1359784d48de48b9d025270985d9,70,0x7f5c764cbc14f9669b88837ca1490cca17c31607,0x94b008aa00579c1307b0ef2c499ad98a8ce58e58,71379460,32037157,32036492
+optimism,curve,1,2023-01-31,0x2a4ff09a9d921f631e4e14a61fc6f7fd7997691ed40acac415c0988fe93f9f96,39,0x7f5c764cbc14f9669b88837ca1490cca17c31607,0x94b008aa00579c1307b0ef2c499ad98a8ce58e58,71426874,16006302,16006705
+optimism,curve,1,2023-01-30,0xf14d489c5cb44772b1082e7589781fac60d0f559fa2999844b2cbb782d434443,75,0x7f5c764cbc14f9669b88837ca1490cca17c31607,0xda10009cbd5d07dd0cecc66161fc93d7c9000da1,71327787,8002330,8002495905917787257
+optimism,curve,1,2023-01-30,0x2615836b17d0b9a467f187a84c497233ee3a815a0ad341d70b63e28d2c6c98df,159,0x7f5c764cbc14f9669b88837ca1490cca17c31607,0xda10009cbd5d07dd0cecc66161fc93d7c9000da1,71331295,128283656,128286430267516232325
+optimism,curve,1,2023-01-30,0x4569f2998b9f00eea1c372d82a36fa32b7aa7ca9fba7378bf63756614e2733e9,3,0x94b008aa00579c1307b0ef2c499ad98a8ce58e58,0xda10009cbd5d07dd0cecc66161fc93d7c9000da1,71338934,255877053,255914112076597632185
+optimism,curve,1,2022-11-12,0xcea94039dcf9ce75c6081f0ed26fc3f6b9193036f901372b002edc67f960a711,9,0x94b008aa00579c1307b0ef2c499ad98a8ce58e58,0x2e3d870790dc77a83dd1d18184acc7439a53f475,37329347,1012,1017988696243052
+optimism,curve,1,2022-05-20,0xf0a56683a1f5c98f734da37362976c19ced781ea2ca1168c203d605593689308,3,0x94b008aa00579c1307b0ef2c499ad98a8ce58e58,0x7f5c764cbc14f9669b88837ca1490cca17c31607,8659786,112531495,106894737
+optimism,curve,1,2022-05-16,0x1daf9b0e5b53bbb0cf2cc82fc56ea4b03e4b04642326c4c34ee5034edcc419a0,5,0x2e3d870790dc77a83dd1d18184acc7439a53f475,0x94b008aa00579c1307b0ef2c499ad98a8ce58e58,8429446,2028066389647480735900,2002456704
+optimism,curve,1,2022-05-11,0x9e35d672d4202923c30f0e65eeaae4b7e922600b2715dacde56a98cef3903062,5,0x94b008aa00579c1307b0ef2c499ad98a8ce58e58,0x2e3d870790dc77a83dd1d18184acc7439a53f475,7910924,1002932210,1001247450302501874441
+optimism,curve,1,2022-05-14,0xa3178cf9778f11a12c896e7967a38ab72b2ca043e2f569dc56e7fe58de6d7995,3,0x2e3d870790dc77a83dd1d18184acc7439a53f475,0xfb21b70922b9f6e3c6274bcd6cb1aa8a0fe20b80,8234971,6057514570698228151262,30000000000
+optimism,curve,1,2023-09-05,0x9e32c44a4f5dae97fe72f8ff3aa52707180beaafb85fc0823e4e273450667ca5,490,0x8c6f28f2f1a3c87f0f938b96d27520d9751ec8d9,0x94b008aa00579c1307b0ef2c499ad98a8ce58e58,109170536,4000860630461955786,3916237747392440611
```

Comment:
I think I may suggest another way of solving this problem, but I am not very sure.
If you search this number `3916237747392440611` in the transaction:
https://optimistic.etherscan.io/tx/0x9e32c44a4f5dae97fe72f8ff3aa52707180beaafb85fc0823e4e273450667ca5

, you can see it is actually swapping to `3.916237747392440611 Curve.fi DAI/USDC/USDT`, and it is not USDT:
https://optimistic.etherscan.io/token/0x1337bedc9d22ecbe766df105c9623922a27963ec

If we can introduce this token, which is not USDT, we may solve this problem perfectly.

Reply:
Hmm, I don't think so. 0x1337BedC9D22ecbe766dF105c9623922A27963EC is [Curve 3pool](https://curve.fi/#/optimism/pools?search=0x1337BedC9D22ecbe766dF105c9623922A27963EC) - not a single token.

Reply:
It's a pool, and a special token at the same time.

Its price feed api:
https://api.coinpaprika.com/v1/coins/3crv-lp-3pool-curve
https://coinpaprika.com/coin/3crv-lp-3pool-curve/

Its etherscan:
https://etherscan.io/token/0x6c3f90f043a72fa612cbac8115ee7e52bde6e490

Moreover, you may not find the exact number matched to amount_raw in these transaction to other tokens.

Reply:
Ok cool, but I'm still not entirely clear how to distinguish token bought from token sold address in this case. How would you set it up in this example: https://dune.com/queries/3365551 (this is comparing results for 1 trade between old and new model using tx you mentioned above)

Reply:
I think this may be the problem in old dex.trades as well, and 3CRV's price is different from USDT now (not 1USD at the current rate).

Reply:
I will first break down the query and see which source these trades come from.

Reply:
Here are the queries:
https://dune.com/queries/3367645
https://dune.com/queries/3367648

Based on the query results(I search the amount_raw in the transaction details), I think all trades from `curvefi_[optimism].MetaPoolSwap_evt_TokenExchangeUnderlying`, are related to `3CRV` trades.

Details:
If a trade is from `curvefi_[optimism].MetaPoolSwap_evt_TokenExchangeUnderlying`, and its token_bought/sold_address is one of `DAI/USDC/USDT`, then that we should change the address to 3CRV's address.


We may only need to fix this part of the code cross all chains for curvefi trades:
https://github.com/duneanalytics/spellbook/blob/main/models/_sector/dex/trades/optimism/platforms/curvefi_optimism_base_trades.sql#L60
```
-- MetaPoolSwap TokenExchangeUnderlying
        SELECT
            'meta' AS pool_type, -- has implications for decimals for curve
            t.evt_block_time AS block_time,
            t.evt_block_number,
            t.buyer AS taker,
            CAST(NULL as varbinary) AS maker,
            -- when amount0 is negative it means taker is buying token0 from the pool
            tokens_bought AS token_bought_amount_raw,
            tokens_sold AS token_sold_amount_raw,
            t.contract_address as project_contract_address,
            t.evt_tx_hash AS tx_hash,
            t.evt_index,
            bought_id,
            sold_id
        FROM {{ source('curvefi_optimism', 'MetaPoolSwap_evt_TokenExchangeUnderlying') }} t
        {% if is_incremental() %}
        WHERE {{ incremental_predicate('t.evt_block_time') }}
        {% endif %}
```

Possible way of fixing: we may move this part outside the `union all`, or use `case` when type is `meta` then change the address.

Reply:
I try to implement a simple fix in #5195 .

Reply:
Ok, there is also extra logic on ethereum spell: `swap_type = 'underlying_exchange_base'` for `token_sold_amount`.

Just for the record tho, I'm not entirely convinced that traders are buying LP tokens in these transactions. I still think it's actually stables swap, but let's see what comes up from your testing.

Reply:
But these trades are underlying exchange events. You can see in one single transaction there are a lot of swap events happened. The final outcomes may be stable swaps, but the middle steps (if we want to include them in the dex.trades), involve this 3CRV stabilizing mechanism.

Reply:
Don't we want to see final outcome in `dex.trades`? I'd say that's what really matters to users/analysts. And that's what's implemented in old lineage too (obv could have been potentially a mistake that no one noticed at the time). I won't pretend that I understand all inner workings of curve tho, so will let you guys to decide where we're going with this.

Reply:
Or we may fix the `amount_raw` to the corresponding tokens', either way we can verify the data in the explorer is ok.

Reply:
No worries. I definitely wouldn't say you're interfering - it's actually good that you're questioning my solutions when you disagree, that's actually helpful! Keeps me in check and discussing is always good to clarify things. Even if we both disagree with each others solutions 😂 And don't worry about your English, it reads perfectly to me and btw English is my second language too.